scripts/nim/utils.mjs

```diff
@@ -0,0 +1,87 @@
+/**
+ * @license Copyright (c) 2003-2025, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-licensing-options
+ */
+
+import upath from 'upath';
+import fs from 'fs-extra';
+import { build } from '@ckeditor/ckeditor5-dev-build-tools';
+import { CKEDITOR5_ROOT_PATH } from '../constants.mjs';
+
+const tsconfig = upath.resolve( CKEDITOR5_ROOT_PATH, 'tsconfig.dist.ckeditor5.json' );
+const banner = upath.resolve( CKEDITOR5_ROOT_PATH, 'scripts/nim/banner.mjs' );
+
+export function dist( path ) {
+	return upath.join( CKEDITOR5_ROOT_PATH, 'dist', path );
+}
+
+export function initializeCKEditor5NpmBuild() {
+	return build( {
+		output: dist( 'ckeditor5.js' ),
+		tsconfig,
+		banner,
+		sourceMap: true,
+		external: [],
+
+		/**
+		 * Because this build runs first, it cleans up the old output folder
+		 * and generates TypeScript declarations and translation files.
+		 * We don't want to repeat this in other steps.
+		 */
+		clean: true,
+		translations: 'packages/**/*.po'
+	} );
+}
```

Comment:
How about adding a `overrides` parameter that is destructured at the end of every function? This way, each function will come with defaults which can be easily overriden.

```suggestion
export function initializeCKEditor5NpmBuild( overrides = {} ) {
	return build( {
		output: dist( 'ckeditor5.js' ),
		tsconfig,
		banner,
		sourceMap: true,
		external: [],

		/**
		 * Because this build runs first, it cleans up the old output folder
		 * and generates TypeScript declarations and translation files.
		 * We don't want to repeat this in other steps.
		 */
		clean: true,
		translations: 'packages/**/*.po',
		...overrides
	} );
}
```

Reply:
Good idea, addressed in https://github.com/ckeditor/ckeditor5/pull/17948/commits/281bfc720399489c8facfe0c4b400b658bc8baed.